src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [itemsPerPage, setItemsPerPage] = useState<number>(5);
+  const [pages, setPages] = useState<number[]>(
+    getNumbers(1, Math.ceil(items.length / itemsPerPage)),
+  );
+  const [itemsIndexOnCurrentPage, setItemsIndexOnCurrentPage] = useState(
+    getNumbers(currentPage, itemsPerPage * currentPage),
+  );
+
+  useEffect(() => {
+    setItemsIndexOnCurrentPage(
+      getNumbers(
+        itemsPerPage * (currentPage - 1),
+        itemsPerPage * currentPage > items.length
+          ? items.length - 1
+          : itemsPerPage * currentPage - 1,
+      ),
+    );
+    setPages(getNumbers(1, Math.ceil(items.length / itemsPerPage)));
+  }, [itemsPerPage, currentPage]);
```

Comment:
looks too complicated, try to simplify this logic without using the states at all here, just save the values in vars